File: Programmers/C++/H-Index.cpp
#include <string>
#include <vector>
#include <algorithm>
#include <iostream>
using namespace std;

int solution(vector<int> citations) {
    int len = citations.size();

    sort(citations.begin(), citations.end());

    for (int i = 0; i < len; i++) {
        if (citations[i] < len - i) continue;
        return len - i;
    }
    return 0;
}

int main() {
    cout << solution({ 3, 0, 6, 1, 5 });

    return 0;
}

File: Programmers/Go/Level2/hindex/hindex.go
package hindex

import "sort"

func Solution(citations []int) int {
	sort.Slice(citations, func(u, v int) bool {
		return citations[u] < citations[v]
	})

	length := len(citations)
	for i := range citations {
		if citations[i] >= length-i {
			return length - i
		}
	}

	return 0
}
